File: colrowsumad.py
m, n = tuple(int(i) for i in input().split())

mat = [list(map(int, input().split())) for _ in range(m)]

colpre = [0]*n
rowpre = [0]*m

for i in range(m):
    for j in range(n):
        rowpre[i] += mat[i][j]
        colpre[j] += mat[i][j]

q = int(input())

for _ in range(q):
    u, v = map(int, input().split())
    u -= 1
    v -= 1
    print(rowpre[u] + colpre[v] - mat[u][v])

File: colrowsumad.cpp
#include<bits/stdc++.h>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(NULL);

    long long m, n;
    cin >> m >> n;

    vector<vector<long long>> mat(m, vector<long long>(n));
    vector<long long> rowpre(m, 0), colpre(n, 0);

    for(long long i = 0; i < m; i++) {
        for(long long j = 0; j < n; j++) {
            cin >> mat[i][j];
            rowpre[i] += mat[i][j];
            colpre[j] += mat[i][j];
        }
    }

    long long q;
    cin >> q;

    while(q--) {
        long long u, v;
        cin >> u >> v;
        u--; v--;
        cout << rowpre[u] + colpre[v] - mat[u][v] << "\n";
    }

    return 0;
}
